nixos/modules/services/misc/tuxclocker.nix

```
@@ -0,0 +1,72 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.tuxclocker;
+in
+{
+  options.programs.tuxclocker = {
+    enable = mkEnableOption (lib.mdDoc ''
+      TuxClocker, a hardware control and monitoring program.
+    '');
+
+    enableAMD = mkEnableOption (lib.mdDoc ''
+      Enable AMD GPU controls.
+      Sets the `amdgpu.ppfeaturemask` kernel parameter to 0xfffd7fff to enable all TuxClocker controls.
+    '');
+
+    enabledNVIDIADevices = mkOption {
+      type = (types.listOf types.int);
+      default = [ ];
+      example = [ 0 1 ];
+      description = lib.mdDoc (''
+        Enable NVIDIA GPU controls for a device by index.
+        Sets the `Coolbits` Xorg option to enable all TuxClocker controls.
+      '');
+    };
+
+    useUnfree = mkOption {
+      type = types.bool;
+      default = true;
+      example = false;
```

```suggestion
      default = false;
      example = true;
```
free by default is a good default for nixos